src/passes/OptimizeInstructions.cpp

```diff
@@ -1290,17 +1290,18 @@ struct OptimizeInstructions
       right->value = Literal::makeSingleZero(type);
       return right;
     }
-    // (signed)x % C_pot != 0   ==>   x & (C_pot - 1) != 0
+    // (signed)x % C_pot != 0   ==>  x & (C_pot - 1) != 0
     {
+      Const* c;
       Binary* inner;
-      if (matches(curr,
-                  binary(Abstract::Ne,
-                         binary(&inner, Abstract::RemS, any(&left), ival()),
-                         ival(0)))) {
-        if (IsPowerOf2((uint64_t)right->value.getInteger())) {
+      if (matches(
+            curr,
+            binary(Abstract::Ne,
+                   binary(&inner, Abstract::RemS, any(&left), constant(&c)),
```

Comment:
`constant` can still be renamed `ival` here for compactness. In general, other than removing `c`, I think all the comments still apply.

Reply:
But I should modify this constant as:
```c++
c->value = c->value.sub(Literal::makeFromInt32(1, left->type));
```
How I can do this without access to `Const*` expression?

Reply:
`ival(&c)` should work fine. `constant` is for when the constant can be either an integer or a floating point constant.

Reply:
I just wondering how about this approach:
```cpp
      int64_t c;
      Binary* inner;
      if (matches(
            curr,
            binary(Abstract::Ne,
                   binary(&inner, Abstract::RemS, any(&left), ival(&c)),
                   ival(0)))) {
        if (IsPowerOf2((uint64_t)c)) {
          if (left->type == Type::i64) {
            optimizePowerOf2URem(inner, (uint64_t)c);
          } else {
            optimizePowerOf2URem(inner, (uint32_t)c);
          }
          return curr;
        }
      }
```
Is it make sense?

Reply:
> constant is for when the constant can be either an integer or a floating point constant.
I see. Thanks!

Reply:
What about rename `constant` to `val` or `xval`?
```
i32()
i64()
f32()
f64()
ival()  ;; i32() | i64()
fval()  ;; f32() | f64()
val()   ;; ival() | fval()
```